modules/auxiliary/scanner/ftp/ftp_version.rb

```diff
@@ -18,31 +18,24 @@ def initialize
       'Author'      => 'hdm',
       'License'     => MSF_LICENSE
     )
-
-    register_options(
-      [
-        Opt::RPORT(21),
-      ], self.class)
   end
 
-  def run_host(target_host)
+  def peer
+    "#{rhost}:#{rport}"
+  end
 
+  def run_host(_target_host)
     begin
-
-    res = connect(true, false)
-
-    if(banner)
-      banner_sanitized = Rex::Text.to_hex_ascii(self.banner.to_s)
-      print_status("#{rhost}:#{rport} FTP Banner: '#{banner_sanitized}'")
-      report_service(:host => rhost, :port => rport, :name => "ftp", :info => banner_sanitized)
-    end
-
-    disconnect
-
+      if connect(true, false) && banner && banner =~ /^220[ -]/
+        recog_banner = banner.gsub(/^220[ -](.*)\r\n/) { "#{Regexp.last_match(1)}\r\n" }.strip
+        info = { "banner" => recog_banner }.merge(recog_info(peer, 'ftp.banner', recog_banner) || {})
+        report_service(host: rhost, port: rport, name: 'ftp', info: info.to_s)
+      end
     rescue ::Interrupt
-      raise $!
+      raise $ERROR_INFO
```

Comment:
Please don't use globals, even English (or special) globals, if possible.  Here, you can do

```ruby
rescue ::Interrupt => interrupt
  raise interrupt
```

and avoid the globals.